File: LeetCode/Backspace String Compare.py
class Solution:
    def backspaceCompare(self, s: str, t: str) -> bool:
        def buildString(string: str) -> str:
            stack = []
            ans = ""
            
            for ch in string:
                if ch != "#":
                    stack.append(ch)
                elif len(stack) > 0:
                    stack.pop()
            
            for ch in stack:
                ans += ch
            
            return ans
        
        return buildString(s) == buildString(t)
            

File: LeetCode/Backspace String Compare.cpp
class Solution {
 public:
  bool backspaceCompare(string s, string t) {
    for (int i = 0; i < s.size(); i++) {
      if (s[i] == '#') {
        s.erase(s.begin() + max(0, i - 1), s.begin() + i + 1);
        i = max(-1, i - 3);
      }
    }

    for (int i = 0; i < t.size(); i++) {
      if (t[i] == '#') {
        t.erase(t.begin() + max(0, i - 1), t.begin() + i + 1);
        i = max(-1, i - 3);
      }
    }

    return s == t;
  }
};